tests/generic/bug01464.phpt

```diff
@@ -0,0 +1,131 @@
+--TEST--
+Test for PHP-1464: GridFS creates unique index with dropDups option
+--SKIPIF--
+<?php require_once "tests/utils/standalone.inc" ?>
+--FILE--
+<?php
+require_once "tests/utils/server.inc";
+
+$host = MongoShellServer::getStandaloneInfo();
+
+$mc = new MongoClient($host);
+$db = $mc->selectDB(dbname());
+
+$filesId = new MongoId('000000000000000000000002');
+$data = new MongoBinData('foo', MongoBinData::BYTE_ARRAY);
+
+// Drop files collection
+$filesCollection = $db->selectCollection('fs.files');
+$filesCollection->drop();
+
+// Drop chunks collection and insert duplicate, orphan chunks
+$chunksCollection = $db->selectCollection('fs.chunks');
+$chunksCollection->drop();
+$chunksCollection->insert(array(
+    '_id' => new MongoId('000000000000000000000002'),
+    'files_id' => new MongoId('000000000000000000000001'),
+    'n' => 0,
+    'data' => new MongoBinData('foo', MongoBinData::BYTE_ARRAY),
+));
+$chunksCollection->insert(array(
+    '_id' => new MongoId('000000000000000000000003'),
+    'files_id' => new MongoId('000000000000000000000001'),
+    'n' => 0,
+    'data' => new MongoBinData('bar', MongoBinData::BYTE_ARRAY),
+));
+
+// Test three methods that start by ensuring the unique index
+echo "MongoGridFS::storeBytes():\n";
+
+try {
+    $db->getGridFS()->storeBytes('foo');
+} catch (MongoGridFSException $e) {
+    echo $e->getMessage(), "\n";
+}
+
+echo "\nMongoGridFS::storeFile():\n";
+
+try {
+    $db->getGridFS()->storeFile(__FILE__);
+} catch (MongoGridFSException $e) {
+    echo $e->getMessage(), "\n";
+}
+
+echo "\nMongoGridFS::remove():\n";
+
+try {
+    $db->getGridFS()->remove();
+} catch (MongoGridFSException $e) {
+    echo $e->getMessage(), "\n";
+}
+
+echo "\nDumping fs.files:\n";
+
+foreach ($filesCollection->find() as $file) {
+    var_dump($file);
+}
+
+echo "\nDumping fs.chunks:\n";
+
+foreach ($chunksCollection->find() as $chunk) {
+    var_dump($chunk);
+}
+
+?>
+==DONE==
+--EXPECTF--
+MongoGridFS::storeBytes():
+Could not store file: %s:%d: %SE11000 duplicate key error index: %s.fs.chunks.$files_id_1_n_1%sdup key: { : ObjectId('000000000000000000000001'), : 0 }
```

Comment:
Testing for the duplicate key exception seemed easier than using stream listeners to watch for inserts to `system.indexes` or `createIndexes` commands, depending on what server version we were dealing with. This exception message pattern should cover 2.4 through 3.0.